chain/chain/src/store.rs

```diff
@@ -1817,26 +1817,49 @@ impl<'a> ChainStoreUpdate<'a> {
             .insert((*hash, shard_id), outgoing_receipts);
     }
 
+    /// For all the outgoing receipts generated in block `hash` at the shards we are tracking
```

Comment:
could you add a unittest for this method - especially on how it behaves on epoch boundary, if we have layout change.

Reply:
I added a unit test. It's hard to add a standalone unit test just for this method, because it requires building a chain history, so I added a new check in the existing sharding upgrade test.

The unit test is more or less just repeating the logic in this function though. I think ideally the test should be more like this: in a setting where we have chunk-only producers enabled, test that we can request the outcome for the receipt from the validators tracking that shard. This won't work now because all validators track all shards. I can create a github issue to track this so we remember to add this check when we test chunk-only producers.